devito/data/data.py

```diff
@@ -179,9 +179,9 @@ def __str__(self):
         return super(Data, self._local).__str__()
 
     @_check_idx
-    def __getitem__(self, glb_idx, comm_type):
+    def __getitem__(self, glb_idx, comm_type, gather=None):
         loc_idx = self._index_glb_to_loc(glb_idx)
-        if comm_type is index_by_index:
+        if comm_type is index_by_index or isinstance(gather, int):
```

Comment:
Also there is a lot of `isinstance(gather, int)`, wouldn't it be nicer to have `gather = int(gather) if isinstance(gather, int) else None` at the beginning and then only needs `if gather` everywhere below. The `int(gather)` probably not necessary actually but just here for safety.

Reply:
`gather` can be, and often is, `0` which is why is instance `isinstance` is used. But we could have the `kwarg` be `gather_rank` and then introduce `gather` as a `True/False` - leads to one more line of code, but code is then a bit tidier I guess.